### DISCUSSION TITLE
In tailwind v4, will vite plugin work in the project that are not vite based? e.g Next JS

### DISCUSSION THREAD

**Discussion by AsuraKev** [tailwindlabs/tailwindcss#13204] [Help]
Hi there,

I was reading about the features that comes with v4. The zero configuration section mentions about leverage vite's module graph. So my question is will v4 work with projects that are not using vite? example will be using it in Next js 14

thanks

**Comment by wongjn** [comment]
The bit about leveraging Vite's module graph is specific to the `@tailwindcss/vite` Vite plugin. I would presume that Tailwind v4 should be able to be integrated into Next.js.

**Reply by AsuraKev** [reply]
thanks :) I saw this line "We’re hoping to expand this approach outside of the Vite ecosystem with other bundler plugins in the future". I guess it ll support it in the future?

**Reply by wongjn** [reply]
I understood that to mean the precise module graph analysis in particular. I would presume that Tailwind v4 should be able to be integrated into Next.js in general, but perhaps not with zero configuration as of right now.

**Reply by AsuraKev** [reply]
thanks for the clarification :)

**Reply by adamwathan** [reply]
Right now for Next you should use the `@tailwindcss/postcss` plugin, will work with zero-configuration still using our crawl-based content detection strategy:

https://tailwindcss.com/blog/tailwindcss-v4-alpha#using-post-css

**Comment by Mac-DevOSYandex** [comment]
Since there is no Next.js plugin in the the pipeline ... Is there a Vite template with Tailwindcss v4 and shadCN/ui. That said, I'll do search, anyways...

**Reply by adamwathan** [reply]
Use the PostCSS plugin with Next, works great!

**Reply by o-az** [reply]
> Use the PostCSS plugin with Next, works great!

any idea if it's possible to use Lightning CSS instead of PostCSS? Since both TailwindCSS v4 and Next.js integrate it.

**Reply by philipp-spiess** [reply]
@o-az The only way to extend Next.js right now in terms of CSS is to use a postcss module, even when using turbopack (which will use lightningcss afterwards to do the minification). We're definitely interested in tighter integrations though and will continue to explore that, but nothing I can share at the moment I'm afraid.
